### DISCUSSION TITLE
How to migrate the drawer to version 3?

### DISCUSSION THREAD

**Discussion by AndreasPB** [saadeghi/daisyui#2290] [Q&A] [answered]
I've been wanting to get on the newest DaisyUI for a while, but any version at or past version 3 breaks my drawer. Anyone that has had to do the same?

I don't mind the overlapping style, but both the height and the way the buttons are handled is a bit unfortunate

Version 2.52.0:
![billede](https://github.com/saadeghi/daisyui/assets/43907402/8da4e0f0-ae18-4778-a686-2aa9c33b64dc)

Version >=3.0.0:
![billede](https://github.com/saadeghi/daisyui/assets/43907402/585bd348-f5f2-453b-a758-17163f84d887)

This is what I've changed for the picture - I have tried a few things but without much luck

```diff
diff --git a/web/src/lib/components/drawer.svelte b/web/src/lib/components/drawer.svelte
index f8891ce..ebb3654 100644
--- a/web/src/lib/components/drawer.svelte
+++ b/web/src/lib/components/drawer.svelte
@@ -1,17 +1,10 @@
 <script lang="ts">
   import { isAuth } from "$lib/stores/auth"
   import DrawerContent from "$lib/components/drawer_content.svelte"
-
-  let drawerToggled: boolean
 </script>
 
-<div class="drawer drawer-mobile">
-  <input
-    id="my-drawer-3"
-    type="checkbox"
-    class="drawer-toggle"
-    bind:checked={drawerToggled}
-  />
+<div class="drawer lg:drawer-open">
+  <input id="drawer" type="checkbox" class="drawer-toggle" />
   <div class="drawer-content flex flex-col h-auto">
     <!-- Page content here -->
     {#if $isAuth}
@@ -26,7 +19,7 @@
     {/if}
   </div>
   <div class="drawer-side">
-    <label for="my-drawer-3" class="drawer-overlay" />
+    <label for="drawer" class="drawer-overlay" />
     <ul class="overflow-y-auto w-80 bg-base-100 border">
       <!-- Sidebar content here -->
       <div class="flex-none">
```

Any help is appreciated :)

**Comment by saadeghi** [comment, marked as answer]
For the height, you can add `min-h-screen` class to `drawer` div.  
In daisyUI 2 we had this div filling the area and stealing the scroll from the body, but that was causing issues. In daisyUI 3 it doesn't do that so the height is the height of your content by default.

About the buttons, it looks like they have a z-index or something that moves them on top. You can add a larger z-index number to `drawer-side` using a class name like `z-[10]` (or a larger number - I don't know what is the z-index of your buttons)

Let me know if you have any questions.

**Reply by AndreasPB** [reply]
That worked! Thank you so much! :) And well done spotting my weird z-indexes (they are now gone :p)
![billede](https://github.com/saadeghi/daisyui/assets/43907402/ee0c5c9f-35c0-46a1-84c0-db07ef7f12f8)
